homeassistant/components/risco/alarm_control_panel.py

```diff
@@ -0,0 +1,133 @@
+
+    def __init__(self, hass, risco, partition):
+        """Init the partition."""
+        self._hass = hass
+        self._risco = risco
+        self._state = partition
+        self._partition_id = partition.id
+
+    @property
+    def device_info(self):
+        """Return device info for this device."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.unique_id,
+            "manufacturer": "Risco",
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique id for that partition."""
+        return f"{self._risco.site_id}_{self._partition_id}"
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self._state.triggered:
+            return STATE_ALARM_TRIGGERED
+        if self._state.arming:
+            return STATE_ALARM_ARMING
+        if self._state.armed:
+            return STATE_ALARM_ARMED_AWAY
+        if self._state.partially_armed:
+            return STATE_ALARM_ARMED_HOME
+        if self._state.disarmed:
+            return STATE_ALARM_DISARMED
+
+        return STATE_UNKNOWN
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return (
+            SUPPORT_ALARM_ARM_HOME
+            | SUPPORT_ALARM_ARM_AWAY
+            | SUPPORT_ALARM_ARM_NIGHT
+            | SUPPORT_ALARM_ARM_CUSTOM_BYPASS
+        )
+
+    @property
+    def code_arm_required(self):
+        """Whether the code is required for arm actions."""
+        return False
+
+    async def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        alarm = await self._risco.disarm(self._partition_id)
+        self._state = alarm.partitions[self._partition_id]
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        alarm = await self._risco.partial_arm(self._partition_id)
+        self._state = alarm.partitions[self._partition_id]
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        alarm = await self._risco.partial_arm(self._partition_id)
+        self._state = alarm.partitions[self._partition_id]
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_custom_bypass(self, code=None):
+        """Send arm custom bypass command."""
+        alarm = await self._risco.partial_arm(self._partition_id)
+        self._state = alarm.partitions[self._partition_id]
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        alarm = await self._risco.arm(self._partition_id)
+        self._state = alarm.partitions[self._partition_id]
+        self.async_write_ha_state()
+
+    async def async_update(self):
+        """Retrieve latest state."""
+        alarm = await self._risco.get_state()
```

Comment:
If I understand the code correctly, the state is used by multiple entities, while this method fetches the state of all at once?

If that is the case, consider implementing the DataUpdateCoordinator that centralizes this call.

Reply:
Yes, using a `DataUpdateCoordinator` is on the roadmap (see the issue description). Wanted to simplify the first pr, and the number of partitions is limited anyway and is mostly 1. In the future I intend to implement binary sensor support for the zones, where you can potentially have dozens or even hundreds, so I'll make sure to implement the coordinator before.